explorer/src/pages/TransactionDetailsPage.tsx

```diff
@@ -75,14 +82,54 @@ function StatusCard({ signature }: Props) {
     [fetchStatus, fetchDetails]
   );
 
+  React.useEffect(() => {
+    return () => {
+      if (autoRefreshTimeout.current) {
+        clearTimeout(autoRefreshTimeout.current);
```

Comment:
Typically it's best to have the clean up callback be in the same effect hook that the timeout is created in. Could you move this there?

Reply:
So used this one for if the user navigates away to clear out the timeout (which is also why I stored the interval id using the ref). I'll message you in chat to discuss.